stac_fastapi/pgstac/core.py

```diff
@@ -114,10 +114,19 @@ async def all_collections(  # noqa: C901
                 )
                 collections_result = {"collections": cols, "links": []}
 
-        linked_collections: List[Collection] = []
-        collections = collections_result["collections"]
-        if collections is not None and len(collections) > 0:
-            for c in collections:
+        collections = Collections(
+            collections=[],
+            links=[],
+            numberReturned=collections_result.get(
+                "numberReturned", len(collections_result["collections"])
+            ),
+        )
+        if collections_result.get("numberMatched", None) is not None:
+            collections["numberMatched"] = collections_result.get("numberMatched")
```

Comment:
we don't return `numberMatched` when pgstac do not return it (when we're not using collection-search)

in Theory we could return numberMatched = numberReturned when not using collection-search 🤷‍♂️